3scale/3scale-operator
# THREESCALE-10388 Migrate apicast DeploymentConfigs to Deployments

# Issue link
JIRA: [THREESCALE-10388](https://issues.redhat.com/browse/THREESCALE-10388)

# What
This is the first PR against the new feature branch `threescale-10163` that will be used while implementing the changes needed to migrate each of 3scale-operator's DeploymentConfigs to Deployments - see [epic](https://issues.redhat.com/browse/THREESCALE-10163).

This PR replaces the `apicast-staging` and `apicast-production` DeploymentConfigs with Deployments. The PR also provides the upgrade scaffolding that will be used by the remaining DeploymentConfigs.

# Verification steps

1. Provision a cluster
2. Install 3scale locally using master
3. Checkout this PR
4. Re-run 3scale locally
5. Verify that the new `apicast-staging` and `apicast-production` Deployments are created
6. Verify that the `apicast-staging` and `apicast-production` DeploymentConfigs are deleted once the Deployments report healthy
7. Login to admin portal and create dummy product/backend.
8. Promote the APIcast configuration to Staging APIcast and curl the staging endpoint with a valid user_key to confirm it's working
9. Promote the APIcast configuration to Production APIcast and curl the production endpoint with a valid user_key to confirm it's working
